km/km_management.c

```diff
@@ -96,14 +97,37 @@ static void* mgt_main(void* arg)
 
 void km_mgt_init(char* path)
 {
-   if (path == NULL) {
-      return;
+   char pipename[128];
+   char* p = getenv(KM_MGTDIR);
+   if (p != NULL) {
+      /*
+       * If KM_MGTDIR is in the environment we ignore the pipe name supplied on the
+       * command line or the KM_MGTPIPE env var.  Instead we use self generated names
+       * in the directory supplied with KM_MGTDIR.
+       */
+      km_mgtdir = strdup(p);
+      if (km_mgtdir == NULL) {
+         km_errx(3, "couldn't allocate KM_MGDIR string");
+      }
+      struct stat sb;
+      int rc = stat(km_mgtdir, &sb);
+      if (rc != 0) {
+         km_err(3, "KM_MGTDIR directory <%s> is not accessible", km_mgtdir);
+      }
+      snprintf(pipename, sizeof(pipename), "%s/kmpipe.%s.%d", km_mgtdir, basename(km_payload_name), getpid());
+      path = pipename;
+   } else {
+      if (path == NULL) {
+         return;
+      }
    }
    if (strlen(path) + 1 > sizeof(addr.sun_path)) {
-      km_errx(2, "mgmt path too long");
+      km_errx(2, "mgmt path <%s> too long, max length %d", path, sizeof(addr.sun_path));
    }
    strncpy(addr.sun_path, path, sizeof(addr.sun_path) - 1);
 
+   km_warnx("snapshot pipe name is %s", path);
```

Comment:
info? I understand this is more for us the the customer, so I thought it makes sense to make it KM_VERBOSE controlled